src/main/java/org/fenixedu/academic/ui/spring/StrutsFunctionalityController.java

```diff
@@ -34,7 +35,10 @@
     @ModelAttribute
     private void selectStrutsFunctionality(HttpServletRequest request) {
         Functionality functionality = RenderersAnnotationProcessor.getFunctionalityForType(getFunctionalityType());
-        BennuPortalDispatcher.selectFunctionality(request,
-                MenuFunctionality.findFunctionality(functionality.getProvider(), functionality.getKey()));
+        MenuFunctionality menuItem = MenuFunctionality.findFunctionality(functionality.getProvider(), functionality.getKey());
+        if (!menuItem.isAvailableForCurrentUser()) {
```

Comment:
Included the null check with the semantic of unauthorized if null, not having the functionality installed in the menu means its unavailable